src/machine/machine_stm32_i2c.go

```diff
@@ -191,27 +169,27 @@ func (i2c I2C) Configure(config I2CConfig) {
 }
 
 func (i2c I2C) Tx(addr uint16, w, r []byte) error {
-	a := address7Bit(addr)
-	if err := i2c.controllerTransmit(a, w); nil != err {
+	// a := address7Bit(addr)
+	if err := i2c.controllerTransmit(addr, w); nil != err {
 		return err
 	}
-	if err := i2c.controllerReceive(a, r); nil != err {
+	if err := i2c.controllerReceive(addr, r); nil != err {
 		return err
 	}
 	return nil
 }
 
-func (i2c I2C) controllerTransmit(addr addressable, w []byte) error {
-
-	if !i2c.waitForFlag(flagBUSY, false) {
-		return errI2CBusReadyTimeout
-	}
+func (i2c I2C) controllerTransmit(addr uint16, w []byte) error {
 
 	// ensure peripheral is enabled
 	if !i2c.Bus.CR1.HasBits(stm32.I2C_CR1_PE) {
 		i2c.Bus.CR1.SetBits(stm32.I2C_CR1_PE)
 	}
 
+	if !i2c.waitForFlag(flagBUSY, false) {
+		return errI2CBusReadyTimeout
+	}
+
```

Comment:
This  is not a pure refactor: these lines might change behavior. Was this intentional? Does this fix an issue? If not, I'd revert these changes unless this change improves something (for example, fixing #1568).

Reply:
These changes were intentional on my part to test if it fixed the problem being reported #1568 (which I can't reproduce), but it sounds like it had no effect. (Doesn't the original implementation order look backwards? It potentially checks for a peripheral flag before the peripheral is enabled....)

Reverted back to original ordering.